### PR TITLE
Prevent NullPointerException in LIVE TestMode

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR resolves an issue when `TestMode.LIVE` is being used and a `RecordNetworkCallPolicy` is added into a pipeline.

### CODE REVIEW

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java [hunk 1]
```diff
@@ -193,7 +191,14 @@ public static boolean shouldClientBeTested(HttpClient client) {
...
 
-    private static TestMode initializeTestMode() {
+    /**
+     * Initializes the {@link TestMode} from the environment configuration {@code AZURE_TEST_MODE}.
+     * <p>
+     * If {@code AZURE_TEST_MODE} isn't configured or is invalid then {@link TestMode#PLAYBACK} is returned.
+     *
+     * @return The {@link TestMode} being used for testing.
+     */
+    public static TestMode initializeTestMode() {
```
Comment (alzimmermsft) [inline sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java:201]: I saw this logic was being used in multiple places so I exposed this as a utility method

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java [hunk 2]
```diff
@@ -110,7 +108,7 @@ public void teardownTest(TestInfo testInfo) {
      *
      * @return The TestMode that has been initialized.
      */
-    public TestMode getTestMode() {
+    public static TestMode getTestMode() {
```
Comment (alzimmermsft) [inline sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java:111]: Made this static as it returns a static value

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java [hunk 3]
```diff
@@ -70,13 +71,18 @@ public RecordNetworkCallPolicy(RecordedData recordedData) {
...
-        Objects.requireNonNull(recordedData, "'recordedData' cannot be null.");
         this.recordedData = recordedData;
         redactor = new RecordingRedactor(redactors);
+
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // Test is running in LIVE mode so it won't be able to record the network call, skip recording code.
+        if (TestBase.getTestMode() == TestMode.LIVE) {
```
Comment (samvaity) [inline sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java:82]: Don't we want to record the new data when running in LIVE mode?

Comment (alzimmermsft) [reply sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java:82]: We record when in `TestMode.RECORD` 😃

**Review by samvaity — APPROVED** [review]

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java]: Reverted as some sub-classes override this (which should be removed)

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java]: Reverted as some sub-classes override this method (which should be removed)

**Pull request merged**